ext/html_tokenizer_ext/parser.c

```diff
@@ -508,6 +508,15 @@ static int parser_document_append(struct parser_t *parser, const char *string, u
   return 1;
 }
 
+static inline void parser_free_scan_string(struct parser_t *parser)
+{
+  if(parser->tk.scan.string) {
+    xfree(parser->tk.scan.string);
+    parser->tk.scan.string = NULL;
+  }
+  return;
+}
```

Comment:
This is freeing one of the tokenizer's things, so it seems like it would be better for it to live in the tokenizer.